runtime/runtime/src/actions.rs

```diff
@@ -436,6 +438,11 @@ pub(crate) fn action_deploy_contract(
     );
     account.set_code_hash(code.get_hash());
     set_code(state_update, account_id.clone(), &code);
+    // Precompile the contract and store result (compiled code or error) in the database.
+    #[allow(unused_must_use)]
+    {
+        precompile_contract(&code, &apply_state.config.wasm_config, apply_state.cache.as_deref());
+    }
```

Comment:
This is usually written as 

```suggestion
    let _ = precompile_contract(&code, &apply_state.config.wasm_config, apply_state.cache.as_deref());
```